src/realm/ucx/bootstrap/bootstrap_mpi.c

```diff
@@ -221,6 +224,23 @@ static int bootstrap_mpi_finalize(bootstrap_handle_t *handle)
 
 int realm_ucp_bootstrap_plugin_init(void *mpi_comm, bootstrap_handle_t *handle)
 {
+  // protect self from unload
+  // for details, please refer to https://github.com/StanfordLegion/realm/issues/331
+  Dl_info info;
+  if(dladdr((void *)&realm_ucp_bootstrap_plugin_init, &info) == 0) {
+    fprintf(stderr, "dladdr failed: cannot find shared object for this function\n");
+    return -1;
+  }
+  if(info.dli_fname == NULL) {
+    fprintf(stderr, "dladdr returned NULL dli_fname\n");
+    return -1;
+  }
+  if(dlopen(info.dli_fname, RTLD_NOW | RTLD_NODELETE) == NULL) {
+    fprintf(stderr, "dlopen failed for %s: %s\n", info.dli_fname, dlerror());
+    return -1;
+  }
+  dlclose(handle);
```

Comment:
Is the intention here to actually close `handle`?

Reply:
My understanding is the dlclose is just to decrease the reference count of the handle.